common-features/data-binding/descriptors.md

```diff
@@ -0,0 +1,322 @@
+
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Filter Mode | FilterDescriptors Property Value | LogicalOperator Property Value |
+| --- | --- | --- |
+| FilterMenu | Two filter descriptor instances per each filtered field. Each filter descriptor instance gets the user input as `Value`. If there is no user input in one of the input fields in the menu then this filter descriptor instance `Value` is null. | Depending on the user choice. |
+| FilterRow | Two filter descriptor instances per each filtered field. The second filter descriptor instance always gets null as `Value`, because there is no second input field. | AND |
+
+
+## Searching
+
+The searching criteria in a Grid or TreeList are stored in an individual `IFilterDescriptor`. Cast it to [`CompositeFilterDescriptor`](#compositefilterdescriptor). The `CompositeFilterDescriptor` holds one child `FilterDescriptor`s for each searchable string column. Each `FilterDescriptor` has the same `Value`, which is the user search input. The value of the `LogicalOperator` property of the `CompositeFilterDescriptor` is `Or`.
+
+
+## Sorting
+
+The sorting criteria are stored in a collection of [`SortDescriptor`](/blazor-ui/api/telerik.datasource.sortdescriptor) objects. Each `SortDescriptor` instance gives access to:
+* The `Member`&mdash;The field where the user sorts.
+* The `SortDirection`&mdash;The sort direction for this sort descriptor.
+
+
+## Grouping
+
+Тhe grouping criteria for each group is stored in an individual collection of [`GroupDescriptor`](/blazor-ui/api/telerik.datasource.groupdescriptor). The `GroupDescriptor` class inherits the `SortDescriptor` class and gives access to the same properties as the `SortDescriptor` class.
+
+
+## Example with OnRead Event Handler
+
+You can obtain the FilterDescriptor, SortDescriptor, GroupDescriptor in the `OnRead` event handler.
+
+````CSHTML
+@using Telerik.DataSource
+@using Telerik.DataSource.Extensions
+
+<p>@ConsoleSim</p>
+
+<TelerikGrid TItem="@SampleData"
+             OnRead="@OnReadHandler"
+             Sortable="true"
+             Groupable="true"
+             FilterMode="@GridFilterMode.FilterMenu"
+             Pageable="true" PageSize="15"
+             Height="400px">
+    <GridToolBarTemplate>
+        <GridSearchBox />
+    </GridToolBarTemplate>
+    <GridColumns>
+        <GridColumn Field="@nameof(SampleData.Id)" />
+        <GridColumn Field="@nameof(SampleData.Name)" />
+        <GridColumn Field="@nameof(SampleData.Team)" />
+        <GridColumn Field="@nameof(SampleData.HireDate)" />
+    </GridColumns>
+</TelerikGrid>
+
+
+@code {
+    private MarkupString ConsoleSim { get; set; } // to showcase what you get
+
+    private async Task OnReadHandler(GridReadEventArgs args)
+    {
+        string output = string.Empty;
+
+        //get the filtering and searching criteria
```

Comment:
Make sure that all comments that represent complete sentences use the appropriate capitalization and punctuation.
```suggestion
        // Get the filtering and searching criteria.
```